sup3r/pipeline/forward_pass.py

```diff
@@ -1587,42 +1605,54 @@ def run(cls, strategy, node_index):
             Index of node on which the forward passes for spatiotemporal chunks
             will be run.
         """
+        start = dt.now()
         if strategy.pass_workers == 1:
             logger.debug(f'Running forward passes on node {node_index} in '
                          'serial.')
-            for chunk_index in strategy.node_chunks[node_index]:
+            for i, chunk_index in enumerate(strategy.node_chunks[node_index]):
+                now = dt.now()
                 cls.incremental_check_run(strategy, node_index, chunk_index)
+                mem = psutil.virtual_memory()
+                logger.info('Finished forward pass on chunk_index='
+                            f'{chunk_index} in {dt.now() - now}. {i + 1} of '
+                            f'{len(strategy.node_chunks[node_index])} '
+                            'complete. Current memory usage is '
+                            f'{mem.used / 1e9:.3f} GB out of '
+                            f'{mem.total / 1e9:.3f} GB total.')
         else:
             logger.debug(f'Running forward passes on node {node_index} in '
                          'parallel with pass_workers='
                          f'{strategy.pass_workers}.')
-            with ThreadPoolExecutor(max_workers=strategy.pass_workers) as exe:
+            with SpawnProcessPool(max_workers=strategy.pass_workers) as exe:
```

Comment:
what kind of compute vs. memory tradeoffs are you seeing here? When the process pool parallelization was after the IO i was seeing no improvement in compute but a penalty in memory (crazy and i cant explain it but it sucked).

Reply:
I saw around a 2x speed up with workers=10 vs workers=1 for 100 chunks with (50,50,10) shape.